src/main/webapp/app/entities/treatment/treatment-dialog.controller.js

```diff
@@ -5,9 +5,9 @@
         .module('vivaxDataManagerApp')
         .controller('TreatmentDialogController', TreatmentDialogController);
 
-    TreatmentDialogController.$inject = ['$scope', '$stateParams', '$uibModalInstance', 'entity', 'Treatment', 'SiteData'];
+    TreatmentDialogController.$inject = ['$scope', '$stateParams', '$uibModalInstance', 'entity', 'Treatment', 'SiteData', 'ShareDataService'];
```

Comment:
You shouldn't change this controller. Leave it as it is oiriginally and then in your form controller have a function like this:

$scope.newInnDrug = function (surveyData) {
                $uibModal.open({
                    templateUrl: 'app/entities/innDrug/innDrug-dialog.html',
                    controller: 'InnDrugDialogController',
                    size: 'lg',
                    resolve: {
                        entity: function () {
                            return {
                                name: null,
                                shortName: null,
                                id: null
                            };
                        }
                    }
                }).result.then(function (result) {
                    $scope.inndrugs.push(result);
                    surveyData.innDrug = result;
                }, function () {
                })
            };